### PR TITLE
Hide progress bar on shallow routing

### LINKED ISSUE #277: Hide progress bar on shallow routing
## Description

On pages with search feature, the url is updated after keyword/filter/sort is changed using shallow routing. 
The progress bar is quite distracting in these pages.

## Expected Behaviour

Progress bar should not show on shallow routing event, especially because loading state for search will be managed separately.

### PR DESCRIPTION
**Pull request by ekamuktia**
Closes #277

## Description

Check if the router event is shallow event, if yes then do not start nprogress

## Current Tasks

- [x] Update router event to start nprogress only for non shallow event
- [x] Test on province, contact, and faq page by changing keyword, filter, and sort
- [x] Add screen recording from testing

Province List
ezgif com-gif-maker

Contact List
ezgif com-gif-maker (1)

FAQ
ezgif com-gif-maker (2)

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -22,8 +22,10 @@ const meta = {
 
 export default function App({ Component, pageProps, router }: AppProps) {
   useEffect(() => {
-    const handleStart = () => {
-      NProgress.start();
+    const handleStart = (_: string, { shallow }: any) => {
```
Comment (resir014) [inline pages/_app.tsx:25]:
```suggestion
    const handleStart = (_: string, { shallow }: { shallow: boolean }) => {
```

Make sure to not use `any`.

**Review by resir014 — CHANGES_REQUESTED** [review]

File: pages/_app.tsx (hunk 1 above)
Comment (ekamuktia) [reply pages/_app.tsx]: Thanks, updated!

**Review by resir014 — APPROVED** [review]
🚢

**Review by mazipan — APPROVED** [review]
Nice ✅

A simple touch

**Pull request merged**